src/gmt_macros.h

```diff
@@ -200,7 +200,7 @@
 
 /*! Determine if we have a special virtual, remote, or query files */
 #define gmt_M_file_is_cache(file) (file != NULL && file[0] == '@' && strncmp (file, "@GMTAPI@-", 9U))
-#define gmt_M_file_is_url(file) (file != NULL && (!strncmp (file, "http:", 5U) || !strncmp (file, "https:", 6U) || !strncmp (file, "ftp:", 4U)))
+#define gmt_M_file_is_url(file) (file != NULL && (!strncmp (file, "http:", 5U) || !strncmp (file, "https:", 6U) || !strncmp (file, "ftp:", 4U) || !strncmp (file, "/vsi", 4U)))
```

Comment:
Test also for WCS & WMS

Reply:
We can continue this tomorrow. file_is_url are looking for regular URLs.  The /vsi and WCS are not regular URLs as I understand it - they need to go via GDAL

Reply:
But if you also are testing for "/vsi" it's not regular URLs anymore.

Reply:
I am confused.  Are you saying that addresses that starts with WCS: or WMS: are regular URLs and we shall pass them to our libcurl download call?  I thought these were special things that must go via GDAL and pertaining to grids.  If I take those out from this function then should we be given that type of file it will end up in our libcurl call.  Please clarify.

Reply:
No. I'm saying that if you test for /vsi (and you are), which are GDAL only, than you should test also for WCS/WMS that are GDAL only too.

Reply:
Great, that is what I thought.  The revised functions does all that:

```
bool gmtlib_found_url_for_gdal (char *fname) {
	/* File names starting as below should not be tested for existence or reading permissions as they
	   are either meant to be read via GDAL.  Regular URLs (e.g., htthp://, ftp:// are read via libcurl. */
	if (!strncmp(fname,"WCS:", 4)  ||
		!strncmp(fname,"WMS:", 4)  ||
	    !strncmp(fname,"/vsi", 4)) {
#ifdef WIN32
		/* On Windows libcurl does not care about the certificates file (see https://github.com/curl/curl/issues/1538)
		   and would fail, so no choice but prevent certificates verification.
		   However, a CURL_CA_BUNDLE=/path/to/curl-ca-bundle.crt will overcome this and will consult the crt file. */
		if (!getenv ("GDAL_HTTP_UNSAFESSL") && !getenv("CURL_CA_BUNDLE"))
			putenv ("GDAL_HTTP_UNSAFESSL=YES");
#endif
		return true;
	}
	else
		return false;
}
```

Good to go?

Reply:
We must be talking different things. The ``gmtlib_found_url_for_gdal `` is fine like that but I'm referring to the ``gmt_M_file_is_url`` macro that you have

#define gmt_M_file_is_url(file) (file != NULL && (!strncmp (file, "http:", 5U) || !strncmp (file, "https:", 6U) || !strncmp (file, "ftp:", 4U) || !strncmp (file, "/vsi", 4U)))